module/Kanbanize/src/Kanbanize/Service/Importer.php

```diff
@@ -0,0 +1,442 @@
+	 */
+	public function importTask($project, $board, Stream $stream, $kanbanizeTask) {
+		$readModelTask = $this->kanbanizeService->findByTaskId($kanbanizeTask['taskid']); //TODO: esplorare nuovi metadati per l'event store
+		if(is_null($readModelTask)){
+			return $this->createTask($kanbanizeTask, $stream);
+		}
+		$task = $this->taskService->getTask($readModelTask->getId());
+		return $this->updateTask($project, $board, $task, $kanbanizeTask);
+	}
+
+	private function updateTask($project, $board, Task $task, $kanbanizeTask){
+		$updatedTask = false;
+		$this->transactionManager->beginTransaction();
+		try {
+			if($task->getAssignee() != $kanbanizeTask['assignee']){
+				$this->updateTaskOwner($task, $kanbanizeTask['assignee']);
+				$updatedTask = true;
+			}
+			$this->transactionManager->commit();
+		}catch (\Exception $e) {
+			$this->transactionManager->rollback();
+			$this->errors[] = "Cannot update task owner of task {subject: {$task->getSubject()}, taskId: {$kanbanizeTask['taskid']}, boardId: {$board['id']}, projectId: {$project['id']}} due to {$e->getMessage()}";
+		}
+		$this->transactionManager->beginTransaction();
+		try{
+			if($task->getColumnName() != $kanbanizeTask['columnname']){
+				$this->updateTaskStatus($task, $kanbanizeTask['columnname']);
+				$updatedTask = true;
+			}
+			$this->transactionManager->commit();
+		}catch (\Exception $e) {
+			$this->transactionManager->rollback();
+			$this->errors[] = "Cannot update task status of task {subject: {$task->getSubject()}, taskId: {$kanbanizeTask['taskid']}, boardId: {$board['id']}, projectId: {$project['id']}} due to {$e->getMessage()}";
+		}
+		$this->transactionManager->beginTransaction();
+		try {
+			if($task->getSubject() != $kanbanizeTask['title']){
+				$task->setSubject($kanbanizeTask['title'], $this->requestedBy);
+				$updatedTask = true;
+			}
+			$this->transactionManager->commit();
+		}catch (\Exception $e) {
+			$this->transactionManager->rollback();
+			$this->errors[] = "Cannot update subject of task {subject: {$task->getSubject()}, taskId: {$kanbanizeTask['taskid']}, boardId: {$board['id']}, projectId: {$project['id']}} due to {$e->getMessage()}";
+		}
+		if($updatedTask){
+			$this->updatedTasks++;
+		}
+		return $task;
+	}
+
+	/**
+	 * 
+	 * @param array $kanbanizeTask
+	 * @param Stream $stream
+	 * @throws Exception
+	 * @return Task
+	 */
+	private function createTask($kanbanizeTask, Stream $stream){
+		$mapping = $this->organization->getSetting('kanbanizeColumnMapping');
+		$status = $mapping[strtoupper($kanbanizeTask['columnname'])];
+		$options = [
+			"taskid" => $kanbanizeTask['taskid'],
+			"columnname" => $kanbanizeTask['columnname'],
+			"status" => $status
+		];
+		$this->transactionManager->beginTransaction();
+		try {
+			$task = Task::create($stream, $kanbanizeTask['title'], $this->requestedBy, $options);
+			$this->taskService->addTask($task);
+			if(!Task::isEmptyAssignee($kanbanizeTask['assignee'])){
```

Comment:
Questa operazione va fatta prima di creare il $task, non in mezzo ad una transazione...